### PR TITLE
chore: Update CHANGELOG

### PR DESCRIPTION
**Pull request by fuzzybinary**
### What and why?

Add missed items and add release notes from native library update.

### Review checklist

- [ ] This pull request has appropriate unit and / or integration tests

### CODE REVIEW

File: packages/Datadog.Unity/CHANGELOG.md [hunk 1]
```diff
@@ -6,9 +6,15 @@
 * Added an option for detecting non-fatal app hangs within a given threshold on iOS.
 * Added documentation tooltips to the Datadog Options window (on Labels).
 * Added support for file / line mappings in C# exceptions through C# Native Stack Mapping
+* Make DatadogWorker's message pools thread safe.
+* Stop Resources in DatadogTrackedWebRequest if the underlying UnityWebRequest is Disposed.
```
Comment (0xnm) [inline packages/Datadog.Unity/CHANGELOG.md:10]:
```suggestion
* Make `DatadogWorker`'s message pools thread safe.
* Stop resources in `DatadogTrackedWebRequest` if the underlying `UnityWebRequest` is disposed.
```

File: packages/Datadog.Unity/CHANGELOG.md [hunk 2]
```diff
@@ -6,9 +6,15 @@
...
 * Updated iOS SDK to 2.17.0
   * Memory warnings are now tracked as RUM errors
   * Fix refresh rate vital for variable refresh rate displays
+* Updated Android SDK to 2.14.0
+  * Increase retry delay on DNS error.
+  * Stop upload worker on upload failure
+  * Ensure UploadWorker uses the SDK instance name.
```
Comment (0xnm) [inline packages/Datadog.Unity/CHANGELOG.md]:
```suggestion
  * Ensure `UploadWorker` uses the SDK instance name.
```

**Review by 0xnm — DISMISSED** [review]

File: packages/Datadog.Unity/CHANGELOG.md (hunk 1 above)
Comment (fuzzybinary) [reply packages/Datadog.Unity/CHANGELOG.md:10]: 🤔 I'm actually split on this. I can see why we might want them to be lowercase, but they are weirdly "proper" nouns referring to Datadog Resources and the specific Dispose method on C#....

Comment (0xnm) [reply packages/Datadog.Unity/CHANGELOG.md:10]: imo they should start from the capital letter if it is a reference to the code symbol (class, interface, etc.) and then be wrapped in the ticks.

**Review by 0xnm — APPROVED** [review]

**Pull request merged**